sequencer/src/block2.rs

```diff
@@ -188,17 +199,43 @@ impl QueryablePayload for BlockPayload {
     type Iter<'a> = Range<Self::TransactionIndex>;
     type InclusionProof = TxInclusionProof;
 
-    fn len(&self, _meta: &Self::Metadata) -> usize {
-        // The number of txs in a block is defined as the minimum of:
-        // (1) the number of txs indicated in the tx table
-        // (2) the number of tx table entries that could fit into the payload
-        // Why? Because (1) could be anything. A block should not be allowed to contain 4 billion 0-length txs.
-        //
-        // The quantity (2) must exclude the first entry of the tx table because this entry indicates only the length of the tx table, not an actual tx.
-        std::cmp::min(
-            self.get_tx_table_len_as().unwrap_or(0),
-            (self.payload.len() / TxTableEntry::byte_len()).saturating_sub(1), // allow space for the tx table length
-        )
+    fn len(&self, meta: &Self::Metadata) -> usize {
+        let entry_len = TxTableEntry::byte_len();
+
+        // The number of nss in a block is defined as the minimum of:
+        // (1) the number of nss indicated in the ns table
+        // (2) the number of ns table entries that could fit inside the ns table byte len
+        // Why? Because (1) could be anything. A block should not be allowed to contain 4 billion 0-length nss.
+        // The quantity (2) must exclude the prefix of the ns table because this prifix indicates only the length of the ns table, not an actual ns.
```

Comment:
Isn't this complexity unnecessary if we're putting the namespace table in the header?